### PR TITLE
fix(op): parse l1 block info from l2 tx input

### LINKED ISSUE #10462: op-reth outputs bad data for "eth_getTransactionReceipt" method
### Describe the bug

I've been testing the op-reth client on base mainnet and found a big difference when comparing the results from the node running on op-reth to the ones from a node running op-geth.
for simplicity, the main difference is the `l1Fee` value, which should be identical and also the reth node is missing some params `l1BaseFeeScalar/l1BlobBaseFee/l1BlobBaseFeeScalar` and also there is `l1FeeScalar` this parameter in op-reth which isn't present in the op-geth result.

### Steps to reproduce

payload for that request: `{"method": "eth_getTransactionReceipt", "params": ["0xca564948e3e825f65731424da063240eec34ba921dd117ac5d06b8c2e0b2d962"], "id": 52, "jsonrpc": "2.0"}`
op-reth service file flags/description:
`ExecStart=/usr/local/bin/op-reth node \
  --ws --ws.port=8546 --ws.addr=0.0.0.0   --ws.origins="*" \
  --http --http.port=8545 --http.addr=0.0.0.0 \
  --http.corsdomain="*" \
  --authrpc.addr=0.0.0.0 \
  --authrpc.jwtsecret=/node/data/secret.jwt \
  --authrpc.port=8551 \
  --datadir=/node/data/reth/ \
  --rollup.sequencer-http=https://mainnet-sequencer.base.org \
  --http.api=admin,debug,eth,net,trace,web3,rpc,reth,ots,eth-call-bundle \
  --ws.api=admin,debug,eth,net,trace,web3,rpc,reth,ots,eth-call-bundle \
  --metrics 0.0.0.0:6060 \
  --chain base \
  --rpc.gascap 18446744073709551615 \
  --rpc.max-connections 429496729 \
  --rpc.max-logs-per-response 0 \
  --rpc.max-subscriptions-per-connection 10000 \
  --rollup.disable-tx-pool-gossip`

### Node logs

```text
`
output from the op-reth client:
{
  "jsonrpc": "2.0",
  "result": {
    "status": "0x1",
    "cumulativeGasUsed": "0xc74f5e",
    "logs": [],
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "type": "0x2",
    "transactionHash": "0xca564948e3e825f65731424da063240eec34ba921dd117ac5d06b8c2e0b2d962",
    "transactionIndex": "0x3e",
    "blockHash": "0x70a8a64a0f8b141718f60e49c30f027cb9e4f91753d5f13a48d8e1ad263c08bf",
    "blockNumber": "0x1185e55",
    "gasUsed": "0x5208",
    "effectiveGasPrice": "0x31b41b",
    "from": "0x889ebdac39408782b5165c5185c1a769b4dd3ce6",
    "to": "0x7449061f45d7b39b3b80b4159286cd8682f60a3c",
    "contractAddress": null,
    "l1Fee": "0x84aaa7",
    "l1FeeScalar": "0.000004",
    "l1GasPrice": "0x50f928b4",
    "l1GasUsed": "0x640"
  },
  "id": 52
}


output from the op-geth client:
{
  "jsonrpc": "2.0",
  "id": 52,
  "result": {
    "blockHash": "0x70a8a64a0f8b141718f60e49c30f027cb9e4f91753d5f13a48d8e1ad263c08bf",
    "blockNumber": "0x1185e55",
    "contractAddress": null,
    "cumulativeGasUsed": "0xc74f5e",
    "effectiveGasPrice": "0x31b41b",
    "from": "0x889ebdac39408782b5165c5185c1a769b4dd3ce6",
    "gasUsed": "0x5208",
    "l1BaseFeeScalar": "0x8dd",
    "l1BlobBaseFee": "0x1",
    "l1BlobBaseFeeScalar": "0x101c12",
    "l1Fee": "0x125f723f3",
    "l1GasPrice": "0x50f928b4",
    "l1GasUsed": "0x640",
    "logs": [],
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "status": "0x1",
    "to": "0x7449061f45d7b39b3b80b4159286cd8682f60a3c",
    "transactionHash": "0xca564948e3e825f65731424da063240eec34ba921dd117ac5d06b8c2e0b2d962",
    "transactionIndex": "0x3e",
    "type": "0x2"
  }
}
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

`op-reth --version
reth Version: 1.0.3
Commit SHA: 390f30aadebcdd509e72cc04327c3b854de076a6
Build Timestamp: 2024-08-05T23:59:17.371843433Z
Build Features: jemalloc,optimism
Build Profile: maxperf`

### What database version are you on?

op-reth db version
2024-08-22T18:23:51.766359Z  INFO Initialized tracing, debug log directory: /root/.cache/reth/logs/optimism
Error: Datadir does not exist: "/root/.local/share/reth/optimism"

Location:
    /root/reth/crates/cli/commands/src/db/mod.rs:71:9

### Which chain / network are you on?

base mainnet

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

source "$HOME/.cargo/env" && RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc,asm-keccak,optimism --bin op-reth

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by emhane**
Closes https://github.com/paradigmxyz/reth/issues/10462

Fixes offsets for parsing l1 base fee scalar and l1 blob base fee scalar

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
the offsets were wrong because of the selector is already removed here?
but doesn't this ignore the block sequencer number as described in the function docs

@clabby mind taking a look at this?

**Comment by mattsse** [comment]
ah maybe the function docs are wrong because the order is

https://github.com/ethereum-optimism/optimism/blob/9452aa66dc7a2e0deb63bca4bf5c7cfc35c3211c/packages/contracts-bedrock/src/L2/L1Block.sol#L129

**Comment by emhane** [comment]
> the offsets were wrong because of the selector is already removed here? 

no they were wrong despite that, base fee scalar was before taken at offset 12..16, when actually it's at offset 0..4. the selector is only 4 bytes.

got the correct order from op-geth code, as linked in the code comment https://github.com/ethereum-optimism/op-geth/blob/60038121c7571a59875ff9ed7679c48c9f73405d/core/types/rollup_cost.go#L317-L328

File: crates/optimism/evm/src/l1.rs [hunk 1]
```diff
@@ -117,14 +124,29 @@ pub fn parse_l1_info_tx_ecotone(data: &[u8]) -> Result<L1BlockInfo, OptimismBloc
         })
     }
 
-    let l1_blob_base_fee_scalar = U256::try_from_be_slice(&data[8..12]).ok_or_else(|| {
+    // https://github.com/ethereum-optimism/op-geth/blob/60038121c7571a59875ff9ed7679c48c9f73405d/core/types/rollup_cost.go#L317-L328
```
Comment (mattsse) [inline crates/optimism/evm/src/l1.rs:127]: let's also reference the contract's function

**Review by mattsse — CHANGES_REQUESTED** [review]
okay, this seems correct, can we update the function docs, because the sequence number is incorrect there

doc nits, pending sanity check @clabby or @refcell

File: crates/optimism/evm/src/l1.rs (hunk 1 above)
Comment (tynes) [reply crates/optimism/evm/src/l1.rs:127]: This is the link you are looking for: https://github.com/ethereum-optimism/optimism/blob/957e13dd504fb336a4be40fb5dd0d8ba0276be34/packages/contracts-bedrock/src/L2/L1Block.sol#L136

Comment (tynes) [reply crates/optimism/evm/src/l1.rs:127]: The docs for `parse_l1_info_tx_ecotone` should be updated

**Comment by tynes** [comment]
This looks good to me given the above comments are addressed

**Review by clabby — APPROVED** [review]
LGTM - looks like the old test for Ecotone also needs to be changed.

**Review by mattsse — APPROVED** [review]
the function docs need to be updated

L111 is incorrect and should be removed

**Pull request merged**